blurring_as_a_service/inference_pipeline/submit_inference_pipeline.py

```diff
@@ -10,58 +12,91 @@
 
 
 @pipeline()
-def inference_pipeline(mounted_root_folder, relative_paths_files_to_blur, model):
-    outputs = BlurringAsAServiceSettings.get_settings()["inference_pipeline"]["outputs"]
+def inference_pipeline():
+    # Iterate over all customers
+    for customer in inference_settings['customers']:
+        customer_name = customer['name']
 
-    detect_and_blur_sensitive_data_step = detect_and_blur_sensitive_data(
-        mounted_root_folder=mounted_root_folder,
-        relative_paths_files_to_blur=relative_paths_files_to_blur,
-        model=model,
-    )
-    detect_and_blur_sensitive_data_step.outputs.results_path = Output(
-        type="uri_folder", mode="rw_mount", path=outputs["results_path"]
-    )
-    detect_and_blur_sensitive_data_step.outputs.yolo_yaml_path = Output(
-        type="uri_folder", mode="rw_mount", path=outputs["yolo_yaml_path"]
-    )
+        container_root_formatted = customer['container_root'].format(
+            subscription=subscription_id,
+            resourcegroup=resource_group,
+            workspace=workspace_name,
+            datastore_name=f"{customer_name}_input_structured"
+        )
+
+        mounted_root_folder = Input(
+            type=AssetTypes.URI_FOLDER,
+            path=container_root_formatted,
+            description="Data to be blurred",
+        )
+
+        files_to_blur_formatted = customer['inputs']['files_to_blur'].format(
+            subscription=subscription_id,
+            resourcegroup=resource_group,
+            workspace=workspace_name,
+            datastore_name=f"{customer_name}_input_structured"
+        )
+
+        relative_paths_files_to_blur = Input(
+            type=AssetTypes.URI_FILE,
+            path=files_to_blur_formatted,
+            description="Data to be blurred",
+        )
+
+        model_parameters = customer['model_parameters']
+        model_parameters_json = json.dumps(model_parameters)  # TODO it seems I can not pass a string to @command_component function
```

Comment:
I think you needed to declare it at Input/Output type, that's why

Reply:
Do I need to write this to a json file? The code now is maybe easier to understand and more efficient

Reply:
What if you read the model parameters from the settings object when you need them?

Reply:
@sdavrieux Like loading the yaml from detect_and_blur_sensitive_data.py?

Im not sure if thats possible. We iterate now over the items (customers) in the yaml file and send the model_parameters for each customer to detect_and_blur_sensitive_data.py. If I load the whole yaml in detect_and_blur_sensitive_data.py, I have to search in all the customer data to find the right model_parameters.